core-bundle/src/Image/Studio/Studio.php

```diff
@@ -0,0 +1,622 @@
+    private const CACHE_META_DATA = 'metaData';
+    private const CACHE_TEMPLATE_DATA = 'templateData';
+    private const CACHE_ORIGINAL_DIMENSIONS = 'dimensions';
+
+    /**
+     * @readonly
+     *
+     * @var ContainerInterface
+     */
+    private $locator;
+
+    /**
+     * Simple internal key-value cache to allow cheap subsequent value access.
+     *
+     * @var array<string, mixed>
+     */
+    private $cache = [];
+
+    /**
+     * The resource's file path.
+     *
+     * @var string|null
+     */
+    private $filePath;
+
+    /**
+     * The resource's file model (if applicable).
+     *
+     * @var FilesModel|null
+     */
+    private $filesModel;
+
+    /**
+     * User defined size configuration.
+     *
+     * @var mixed|null
+     */
+    private $sizeConfiguration;
+
+    /**
+     * User defined custom locale.
+     *
+     * @var string|null
+     */
+    private $locale;
+
+    /**
+     * User defined meta data.
+     *
+     * @var MetaData|null
+     */
+    private $metaData;
+
+    /**
+     * Determines if a secondary image (e.g. light box) will be created (if applicable).
+     *
+     * @var bool
+     */
+    private $allowSecondary;
+
+    public function __construct(ContainerInterface $locator)
+    {
+        $this->locator = $locator;
+    }
+
+    /**
+     * Set the image resource.
+     *
+     * @param string|FilesModel $identifier can be a FilesModel, a tl_files uuid/id/path or a file system path
+     */
+    public function from($identifier): self
+    {
+        $this->filePath = $this->locateResource($identifier, $this->filesModel);
+
+        $this->invalidateCache(self::CACHE_PICTURE);
+
+        return $this;
+    }
+
+    /**
+     * Set a size configuration that will be applied to the resource.
+     *
+     * @param int|string|array|PictureConfiguration $size a picture size configuration or reference
+     */
+    public function setSize($size): self
+    {
+        $this->sizeConfiguration = $size;
+
+        $this->invalidateCache(self::CACHE_PICTURE);
+
+        return $this;
+    }
+
+    /**
+     * Set custom meta data. By default or if the argument is set to null, meta
+     * data from the file model (if available) will be used instead.
+     */
+    public function setMetaData(?MetaData $metaData): self
+    {
+        $this->metaData = $metaData;
+
+        return $this;
+    }
+
+    /**
+     * Set a custom locale. By default or if the argument is set to null, the
+     * locale is determined from the request context and/or system settings.
+     */
+    public function setLocale(?string $locale): self
+    {
+        $this->locale = $locale;
+
+        $this->invalidateCache(self::CACHE_META_DATA);
+
+        return $this;
+    }
+
+    /**
+     * Enable/disable the creation of secondary image content (e.g. a light box
+     * or fullsize image). This setting is disabled by default.
+     */
+    public function allowSecondary(bool $allow = true): self
```

Comment:
Haha, I had named it `enableSecondary` before and changed it because i found `enableSecondary(false)` a bit strange. But I agree, `allow` is not the best - I changed it to `createSecondary` for now.

Reply:
Changed my mind again, and called it `enableSecondary()` again. :smile: This is less confusing together with `createDerived()`.